### DISCUSSION TITLE
Considering the use of DCAT-AP to describe datasets

### DISCUSSION THREAD

**Discussion by EmidioStani** [openml/openml-data#48] [Ideas]
Hello,

OpenML provides already metadata for datasets which can be obtained via API.
Have you considered the use of DCAT/DCAT-AP for the descriptions of datasets ?
Already portals like the European Data Portal make use of it

**Comment by joaquinvanschoren** [comment]
Hi Emidio, 

Can you share examples of datasets described with DCAT-AP? I would expect that we already cover most of it? 
Or do you mean an RDF export? 

We are planning to update descriptions anyway for version 2 of the API. There are multiple ways to describe ML datasets, and we are also looking at datasheets: https://arxiv.org/abs/1803.09010

**Comment by EmidioStani** [comment]
Hello @joaquinvanschoren ,

The JSON provided by openml API is a flat structure while it could be nice to make it more hierarchical (splitting a dataset from distribution, etc, see [ DCAT-AP data model](https://github.com/SEMICeu/DCAT-AP/blob/master/releases/2.1.0/dcat-ap_2.1.0.png).

The minimum you could do, keeping the flat structure, is having something like this (taking credit-g json from: https://www.openml.org/api/v1/json/data/31):

{
	"@context":  [
		"https://raw.githubusercontent.com/SEMICeu/DCAT-AP/master/releases/2.1.0/dcat-ap_2.1.0.jsonld",
		{"id": "@id",
		 "type": "@type",
                "name": "http://purl.org/dc/terms/title",
               "processing_date": "http://purl.org/dc/terms/issued",
                "@base":"https://www.openml.org/"
		}
	],
	"type": "Dataset",
	"id": "dataset/31",
	"name": "credit-g",
	"version": "1",
	"description": "**Author**: Dr. Hans Hofmann  \n**Source**: [UCI](https://archive.ics.uci.edu/ml/datasets/statlog+(german+credit+data)) - 1994    \n**Please cite**: [UCI](https://archive.ics.uci.edu/ml/citation_policy.html)\n\n**German Credit dataset**  \nThis dataset classifies people described by a set of attributes as good or bad credit risks.\n\nThis dataset comes with a cost matrix: \n``` \nGood  Bad (predicted)  \nGood   0    1   (actual)  \nBad    5    0  \n```\n\nIt is worse to class a customer as good when they are bad (5), than it is to class a customer as bad when they are good (1).  \n\n### Attribute description  \n\n1. Status of existing checking account, in Deutsche Mark.  \n2. Duration in months  \n3. Credit history (credits taken, paid back duly, delays, critical accounts)  \n4. Purpose of the credit (car, television,...)  \n5. Credit amount  \n6. Status of savings account/bonds, in Deutsche Mark.  \n7. Present employment, in number of years.  \n8. Installment rate in percentage of disposable income  \n9. Personal status (married, single,...) and sex  \n10. Other debtors / guarantors  \n11. Present residence since X years  \n12. Property (e.g. real estate)  \n13. Age in years  \n14. Other installment plans (banks, stores)  \n15. Housing (rent, own,...)  \n16. Number of existing credits at this bank  \n17. Job  \n18. Number of people being liable to provide maintenance for  \n19. Telephone (yes,no)  \n20. Foreign worker (yes,no)",
	"description_version": "4",
	"format": "format/ARFF",
	"creator": "Dr. Hans Hofmann",
	"collection_date": "1994-11-17",
	"upload_date": "2014-04-06T23:21:47",
	"language": "language/English",
	"licence": "license/Public",
	"url": "https://old.openml.org/data/v1/download/31/credit-g.arff",
	"file_id": "31",
	"default_target_attribute": "class",
	"version_label": "1",
	"citation": "https://archive.ics.uci.edu/ml/citation_policy.html",
	"tag": [
		"credit_scoring",
		"finance_problem",
		"mythbusting_1",
		"OpenML-CC18",
		"OpenML100",
		"study_1",
		"study_123",
		"study_14",
		"study_144",
		"study_15",
		"study_20",
		"study_218",
		"study_241",
		"study_34",
		"study_37",
		"study_41",
		"study_50",
		"study_52",
		"study_7",
		"study_70",
		"study_98",
		"study_99",
		"uci"
	],
	"visibility": "public",
	"original_data_url": "https://archive.ics.uci.edu/ml/datasets/statlog+(german+credit+data)",
	"paper_url": "https://dl.acm.org/doi/abs/10.1145/967900.968104",
	"minio_url": "http://openml1.win.tue.nl/dataset31/dataset_31.pq",
	"status": "status/active",
	"processing_date": "2020-11-20 19:44:53",
	"md5_checksum": "9a475053fed0c26ee95cd4525e50074c"
}

you can test by copy and paste on the [JSON-LD playground](https://json-ld.org/playground/) which generate in the N-quads tab the triples.

Then you can validate the triples, against the [DCAT-AP SHACL shapes](https://github.com/SEMICeu/DCAT-AP/blob/55617123a940d00f781cc9204b572f9ab46776e1/releases/2.1.0/dcat-ap_2.1.0_shacl_shapes.ttl) with the zazuko shacl playground on: https://shacl-playground.zazuko.com/

As you can see I renamed the id, language, license, format to make them better uri, otherwise will be relative to the base url, the best would be for format using IANA mediatype or Uri and for language use the Uri from publications office

The best would be make the structure more hierarchical, distinguishing dataset from distribution, put creator in a separate structure, etc.

I think you can combine the considerations of the paper with DCAT-AP
